compiler/qsc/src/interpret.rs

```diff
@@ -97,6 +98,9 @@ pub enum Error {
     #[diagnostic(code("Qsc.Interpret.NotAnOperation"))]
     #[diagnostic(help("provide the name of a callable or a lambda expression"))]
     NotAnOperation,
+    #[error("partial evaluation error")]
+    #[diagnostic(transparent)]
+    PartialEvaluation(#[from] qsc_partial_eval::Error),
```

Comment:
Does this need to be a `WithSource` so that spans get mapped correctly?

Reply:
PartialEval doesn't have the source. It can be updated in the future, but won't work right now.